File: bench/Reference/ackermann.erl
#!/usr/bin/env escript
%% Ackermann 3 11

main(_) ->
    io:format("~p~n", [ack(3, 11)]).

ack(0, N) -> N + 1;
ack(M, 0) -> ack(M - 1, 1);
ack(M, N) -> ack(M - 1, ack(M, N - 1)).

File: bench/Reference/ackermann.java
class ackermann {
    static long ack(long m, long n) {
        if (m == 0) return n + 1;
        if (n == 0) return ack(m - 1, 1);
        return ack(m - 1, ack(m, n - 1));
    }
    public static void main(String[] args) { System.out.println(ack(3, 10)); }
}
